filebeat/prospector/reloader.go

```diff
@@ -0,0 +1,167 @@
+package prospector
+
+import (
+	"expvar"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/elastic/beats/filebeat/registrar"
+	"github.com/elastic/beats/libbeat/cfgfile"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+var (
+	debugr            = logp.MakeDebug("filebeat.reloader")
+	configReloads     = expvar.NewInt("filebeat.config.reloads")
+	prospectorStarts  = expvar.NewInt("filebeat.config.prospector.starts")
+	prospectorStops   = expvar.NewInt("filebeat.config.prospector.stops")
+	prospectorRunning = expvar.NewInt("filebeat.config.prospector.running")
```

Comment:
This seems to only count dynamically configured prospectors, right? We should reflect that in the name.